### PR TITLE
Feature/add component tests

### PR DESCRIPTION
**Pull request by HeJones**
### What

To enable features tests to be added to this service the following things have been done: 
- Add component tests to Make File
- Add features package to hold feature tests
- Added example tests to test the hello world endpoint - one is with a custom step, and one uses a step in the library
- Add ci files

### How to review

If you run `make test-component` the feature tests pass

### Who can review
Anyone

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,3 +38,7 @@ test:
 	go test -cover -race ./...
 
 .PHONY: test build debug
+
+.PHONY: test-component
+test-component:
+	go test -cover -coverpkg=github.com/ONSdigital/dp-identity-api/... -component
```
Comment (nshumoogum) [inline Makefile]: Probably make sense to add -race flag after -cover?

File: ci/component.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.15.8
```
Comment (nshumoogum) [inline ci/component.yml]: Bump to 1.16.3?

File: go.mod [hunk 3]
```diff
@@ -4,10 +4,12 @@ go 1.15
```
Comment (nshumoogum) [inline go.mod:4]: go.mod file should be updated to latest go version 1.16

File: main_test.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+	ctx.AfterScenario(func(*godog.Scenario, error) {
+		_ = component.Close()
+	})
+
+	component.RegisterSteps(ctx)
+}
+
+func (f *ComponentTest) InitializeTestSuite(ctx *godog.TestSuiteContext) {
+
```
Comment (nshumoogum) [inline main_test.go]: Consider removing empty whitespace

File: main_test.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+var componentFlag = flag.Bool("component", false, "perform component tests")
+
+type ComponentTest struct {
+	MongoFeature *componenttest.MongoFeature
+}
+
+func (f *ComponentTest) InitializeScenario(ctx *godog.ScenarioContext) {
+	component, err := steps.NewComponent()
+	if err != nil {
+		panic(err)
```
Comment (nshumoogum) [inline main_test.go]: Would it not make more sense to use os.Exit(1)? And include a log line prior to exiting to log out the error?

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: ci/component.yml (hunk 2 above)
Comment (nshumoogum) [reply ci/component.yml]: Probably want to bump tagged version in unit.yml and build.yml

**Comment by nshumoogum** [comment]
github.com/gogo/protobuf - needs updating to use latest version 1.3.2 instead of 1.3.1 - hopefully this will resolve audit issues

Using `go get -u ./...; go mod tidy`

File: main_test.go (hunk 5 above)
Comment (jon-acker) [reply main_test.go]: yes, panic should def be removed, it would stop the test suite :)

File: features/helloworld.feature [hunk 6]
```diff
@@ -0,0 +1,14 @@
+Feature: Helloworld
+
+
+
```
Comment (mike-kelly-methods) [inline features/helloworld.feature]: Excessive number of newlines.

**Review by mike-kelly-methods — APPROVED** [review]
👍

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**